File: src/main/java/net/sjrx/intellij/plugins/systemdunitfiles/filetypes/DeviceFileType.java
package net.sjrx.intellij.plugins.systemdunitfiles.filetypes;

import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileIcon;
import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileLanguage;
import org.jetbrains.annotations.Nls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.Icon;

public class DeviceFileType extends AbstractUnitFileType {
  public static final DeviceFileType INSTANCE = new DeviceFileType();

  private DeviceFileType() {
    super(UnitFileLanguage.INSTANCE);
  }

  @NotNull
  @Override
  public String getName() {
    return "Device unit configuration for systemd";
  }

  @NotNull
  @Override
  public String getDescription() {
    return getDisplayName();
  }

  @NotNull
  @Override
  public String getDefaultExtension() {
    return "device";
  }

  @Nullable
  @Override
  public Icon getIcon() {
    return UnitFileIcon.FILE;
  }
  
  @Nls
  @Override
  public @NotNull String getDisplayName() {
    return "Device unit configuration (systemd)";
  }
}

File: src/main/kotlin/net/sjrx/intellij/plugins/systemdunitfiles/filetypes/DeviceFileType.kt
package net.sjrx.intellij.plugins.systemdunitfiles.filetypes

import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileIcon
import net.sjrx.intellij.plugins.systemdunitfiles.UnitFileLanguage
import org.jetbrains.annotations.Nls
import javax.swing.Icon

class DeviceFileType private constructor() : AbstractUnitFileType(UnitFileLanguage.INSTANCE) {
  override fun getName(): String {
    return "Device unit configuration for systemd"
  }

  override fun getDescription(): String {
    return displayName
  }

  override fun getDefaultExtension(): String {
    return "device"
  }

  override fun getIcon(): Icon? {
    return UnitFileIcon.DEVICE
  }

  override fun getDisplayName(): @Nls String {
    return "Device unit configuration (systemd)"
  }

  companion object {
    val INSTANCE = DeviceFileType()
  }
}
